### PR TITLE
refactor(core): centAcl optimize memory and time execution

### PR DESCRIPTION
**Pull request by kduret**
## Description

With that patch, centacl uses only ID (old legacy code for nagios centacl is removed). And also the script is faster to execute.

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [x] 19.10.x (master)

### CODE REVIEW

File: cron/centAcl-Func.php [hunk 1]
```diff
@@ -128,36 +128,35 @@ function getCentAclRunningState()
  * @param integer $res_id
```
Comment (jdelpierre) [inline cron/centAcl-Func.php]: $resId

File: cron/centAcl-Func.php [hunk 2]
```diff
@@ -128,36 +128,35 @@ function getCentAclRunningState()
...
-    $dbResult->bindValue(':resId', $res_id, \PDO::PARAM_INT);
+    $dbResult->bindValue(':resId', $resId, \PDO::PARAM_INT);
     $dbResult->execute();
 
     if ($dbResult->rowCount()) {
         $host = array();
         while ($row = $dbResult->fetch()) {
             if (isset($hostTmp[$row['host_host_id']])) {
-                $host[$row['host_host_id']] = $hostCache[$row['host_host_id']];
+                $host[$row['host_host_id']] = 1;
```
Comment (jdelpierre) [inline cron/centAcl-Func.php]: Why you fix the value "1"?

File: cron/centAcl-Func.php [hunk 3]
```diff
@@ -178,40 +177,40 @@ function getFilteredPollers($host, $res_id)
  * @param integer $res_id
```
Comment (jdelpierre) [inline cron/centAcl-Func.php]: $resId

File: cron/centAcl-Func.php [hunk 4]
```diff
@@ -233,27 +232,28 @@ function getFilteredHostCategories($host, $res_id)
...
```
Comment (jdelpierre) [inline cron/centAcl-Func.php]: $resId

File: cron/centAcl-Func.php [hunk 5]
```diff
@@ -262,19 +262,19 @@ function getAuthorizedCategories($res_id)
  * @param integer $service_id
```
Comment (jdelpierre) [inline cron/centAcl-Func.php]: $serviceId

File: cron/centAcl-Func.php [hunk 6]
```diff
@@ -233,27 +232,28 @@ function getFilteredHostCategories($host, $res_id)
  * @param integer $res_id
  * @return array $tab_categories
```
Comment (jdelpierre) [inline cron/centAcl-Func.php]: $tabCategories

File: cron/centAcl-Func.php [hunk 7]
```diff
@@ -334,19 +334,19 @@ function getACLSGForHost($pearDB, $host_id, $resId)
  * @param int $res_id The ACL resource id
```
Comment (jdelpierre) [inline cron/centAcl-Func.php]: $resId

File: cron/centAcl-Func.php [hunk 8]
```diff
@@ -394,19 +394,19 @@ function hasHostCategoryFilter($res_id)
...
```
Comment (jdelpierre) [inline cron/centAcl-Func.php]: $resId

File: cron/centAcl-Func.php [hunk 9]
```diff
@@ -418,55 +418,55 @@ function hasServiceCategoryFilter($res_id)
...
-        if ($tab_svc) {
-            foreach ($tab_svc as $svc_descr => $svc_id) {
-                $tab = getServiceTemplateCategoryList($svc_id);
+        if ($tabSvc) {
+            foreach (array_keys($tabSvc) as $serviceId) {
+                $tab = getServiceTemplateCategoryList($serviceId);
                 foreach ($tab as $t) {
                     if (isset($authorizedCategories[$t])) {
-                        $tab_services[$svc_descr] = $svc_id;
+                        $tabServices[$serviceId] = 1;
```
Comment (jdelpierre) [inline cron/centAcl-Func.php]: Why you fix the value "1"?

File: cron/centAcl-Func.php [hunk 10]
```diff
@@ -534,14 +531,14 @@ function getMyHostServicesByName($host_id = null)
...
-    $sql = "SELECT meta_id FROM acl_resources_meta_relations WHERE acl_res_id = {$db->escape($resId)}";
+    $sql = "SELECT meta_id FROM acl_resources_meta_relations WHERE acl_res_id = " . $resId;
     $res = $db->query($sql);
     $arr = array();
     if ($res->rowCount()) {
         $hostId = $metaObj->getRealHostId();
         while ($row = $res->fetch()) {
             $svcId = $metaObj->getRealServiceId($row['meta_id']);
-            $arr['_Module_Meta']['meta_' . $row['meta_id']] = "$hostId,$svcId";
+            $arr[$hostId][$svcId] = 1;
```
Comment (jdelpierre) [inline cron/centAcl-Func.php]: Why you fix the value "1"?

File: cron/centAcl.php [hunk 11]
```diff
@@ -377,70 +374,62 @@
...
+            "SELECT host_host_id, acl_res_id
+            FROM acl_resources_host_relations"
         );
         while ($h = $dbResult->fetch()) {
             if (!isset($hostIncCache[$h["acl_res_id"]])) {
-                $hostIncCache[$h["acl_res_id"]] = array();
+                $hostIncCache[$h["acl_res_id"]] = [];
             }
-            $hostIncCache[$h["acl_res_id"]][$h["host_id"]] = $h["host_name"];
+            $hostIncCache[$h["acl_res_id"]][$h["host_host_id"]] = 1;
```
Comment (jdelpierre) [inline cron/centAcl.php]: Why you fix the value "1"?

Comment (kduret) [reply cron/centAcl.php]: I guess cause isset is better (in performance) than in_array

File: cron/centAcl-Func.php (hunk 10 above)
Comment (kduret) [reply cron/centAcl-Func.php]: I guess cause isset is better (in performance) than in_array

File: cron/centAcl-Func.php (hunk 9 above)
Comment (kduret) [reply cron/centAcl-Func.php]: I guess cause isset is better (in performance) than in_array

File: cron/centAcl-Func.php (hunk 2 above)
Comment (kduret) [reply cron/centAcl-Func.php]: I guess cause isset is better (in performance) than in_array

File: cron/centAcl-Func.php [hunk 12]
```diff
@@ -534,14 +531,14 @@ function getMyHostServicesByName($host_id = null)
  */
 function getMetaServices($resId, $db, $metaObj)
 {
-    $sql = "SELECT meta_id FROM acl_resources_meta_relations WHERE acl_res_id = {$db->escape($resId)}";
+    $sql = "SELECT meta_id FROM acl_resources_meta_relations WHERE acl_res_id = " . $resId;
```
Comment (callapa) [inline cron/centAcl-Func.php]:
```suggestion
    $sql = "SELECT meta_id FROM acl_resources_meta_relations WHERE acl_res_id = " . (int) $resId;
```

**Pull request merged**